src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1464,178 @@ def write_sorting(
             write_waveforms=True,
         )
     return nwbfile_out
+
+
+def write_waveforms(
+    waveform_extractor: WaveformExtractor,
+    nwbfile_path: OptionalFilePathType = None,
+    nwbfile: Optional[pynwb.NWBFile] = None,
+    metadata: Optional[dict] = None,
+    recording: Optional[BaseRecording] = None,
```

Comment:
Minor but overwrite should go after metadata to keep to keep the signature consistent. That is, we should swap recording with overwrite here.